webapp/src/ts/modules/contacts/contacts-edit.component.ts

```diff
@@ -299,6 +376,152 @@ export class ContactsEditComponent implements OnInit, OnDestroy, AfterViewInit {
     };
   }
 
+  private parseXmlForm(form): Document | undefined {
+    // Below line is taken from enketo-core form-model line 114
+    const xmlDoc = this.parser.parseFromString(form.getDataStr({ irrelevant: false }), 'text/xml');
+    if (xmlDoc.getElementsByTagName('parsererror').length > 0) {
+      console.error('Error parsing XML:', xmlDoc.getElementsByTagName('parsererror')[0].textContent);
+      this.globalActions.setEnketoError('Error parsing xml document');
+      return;
+    } 
+    return xmlDoc; 
+  }
+
+  private getFormValues(xmlDoc: Document, propPaths: string[]){
+    const values:string[] = [];
+    for (const path of propPaths){
+      try {
+        const value = xmlDoc.evaluate(
+          path,
+          xmlDoc,
+          null,
+          XPathResult.STRING_TYPE,
+          null
+        ).stringValue;
+        values.push(value);
+      } catch (error) {
+        console.error(`Path ${path} value could not be resolved!`);
+      }
+    }
+    return values;
+  }
+
+  private queryPropInfo(xmlDoc: Document, info: Exclude<QueryParamsInfoType, undefined>){
+    const propPaths = info.valuePaths;
+    const values = this.getFormValues(xmlDoc, propPaths);
+    const result = info.query(...values);
+    return result;
+  }
+
+  private calculateSiblingLikenessScore(sibling: Sibling, props: { value: string; compareToDocProp: string }[]) {
+    let totalScore = 0;
+    let count = 0;
+
+    for (const field of props) {
+      if (Object.prototype.hasOwnProperty.call(sibling, field.compareToDocProp)) {
+        const test = this.strategy(field.value.toLowerCase(), sibling[field.compareToDocProp].toLowerCase());
+        totalScore += test;
+        count++;
+      }
+    }
+
+    return count > 0 ? totalScore / count : null;
```

Comment:
Does the alternative return type need to be `null` or is there an appropriate default value of type `int`?